src/main/java/com/minecolonies/coremod/commands/CommandEntryPointNew.java

```diff
@@ -1,573 +1,88 @@
-        super();
-        root = buildMenu(NavigationMenuType.MINECOLONIES);
-    }
-
-    /**
-     * Removed so we can control this at the config
-     * Only allow OP's execute the commands.
-     *
-     * @return the int permission level (3 for OP).
-     * public int getRequiredPermissionLevel(){return OP_PERMISSION_LEVEL;}
-     */
-    @NotNull
-    @Override
-    public List<String> getAliases()
-    {
-        return Arrays.asList("mc", "col", "mcol", "mcolonies", "minecol", "minecolonies");
-    }
-
-    private TreeNode<IMenu> buildMenu(@NotNull final IMenuType menuType)
-    {
-        return buildMenu(menuType, new HashSet<>());
-    }
-
-    private TreeNode<IMenu> buildMenu(@NotNull final IMenuType menuType, @NotNull final Set<IMenuType> menuTypesSoFar)
-    {
-        // Prevent recursion
-        if (menuTypesSoFar.contains(menuType))
-        {
-            return null;
-        }
-        else
+        public boolean hasPermission(final ForgePermissionNodes forgePermissionNode, final EntityPlayer player)
         {
-            menuTypesSoFar.add(menuType);
+            return PermissionAPI.hasPermission(player.getGameProfile(), forgePermissionNode.getNodeName(), new PlayerContext(player));
         }
 
-        final IMenu menu = menuType.getMenu();
-        final TreeNode<IMenu> treeNode = new TreeNode<>(menu);
-
-        if (menuType.isNavigationMenu())
+        @Override
+        public boolean canUseCommands(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender)
         {
-            final NavigationMenu navigationMenu = (NavigationMenu) menu;
-            for (final IMenuType childMenuType : navigationMenu.getChildrenMenuList())
+            if (sender instanceof EntityPlayer)
             {
-                final TreeNode<IMenu> childTreeNode = buildMenu(childMenuType, menuTypesSoFar);
-                treeNode.addChild(childTreeNode);
+                return AbstractSingleCommand.isPlayerOpped(sender) || Configurations.gameplay.opLevelForServer <= 0;
             }
+            return true;
         }
-
-        return treeNode;
     }
 
-    @NotNull
-    @Override
-    public String getName()
+    protected ForgeCommandsPermissionsChecker getPermissionsChecker()
     {
-        return root.getData().getMenuItemName();
+        return new ForgeCommandsPermissionsChecker();
     }
 
-    @NotNull
-    @Override
-    public String getUsage(final ICommandSender sender)
+    protected NavigationMenuType getRootNavigationMenuType()
     {
-        return getCommandUsage(sender, root);
+        return NavigationMenuType.MINECOLONIES;
     }
-
-    @NotNull
-    private static String getCommandUsage(final ICommandSender sender, final TreeNode<IMenu> currentMenuTreeNode)
+    
+    protected ModuleContext getModuleContext()
     {
-        final IMenuType currentMenuType = currentMenuTreeNode.getData().getMenuType();
-        if (currentMenuTreeNode.hasChildren())
+        return new ModuleContext()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1604)